src/models.py

```diff
@@ -965,10 +965,21 @@ def _seq_gen_forward(self, batch, task, predict):
 
         return out
 
-    def _tagger_forward(self, batch, task, predict):
-        ''' For sequence tagging '''
+    def _tagger_forward(self, batch: dict, task: TaggingTask, predict: bool) -> dict:
+        '''
+        This function is for sequence tagging (one-to-one mapping between words and tags).
+        Args:
+                batch: a dict of inputs and target tags 
+                task: TaggingTask
+                predict: (boolean) predict mode (not supported)
+        Returns
+            out: (dict)
+                - 'logits': output layer, dimension: [batchSize * task.max_seq_len, task.num_tags]
+                - 'loss': size average CE loss
+        '''
         out = {}
-        b_size, seq_len, _ = batch['inputs']['elmo'].size()
+        # batch[inputs] only has one item
+        b_size, seq_len = list(batch["inputs"].values()[0]).size()
```

Comment:
Assuming this works both with and without ELMo?

Reply:
Yep!

Reply:
Parens are off? (`list.size()` doesn't make sense, does it?)